src/main/kotlin/com/fasterxml/jackson/module/kotlin/KotlinModule.kt

```diff
@@ -54,6 +54,13 @@ class KotlinModule @Deprecated(
     val singletonSupport: SingletonSupport = DISABLED,
     val strictNullChecks: Boolean = false
 ) : SimpleModule(KotlinModule::class.java.name, PackageVersion.VERSION) {
+    init {
+        if (!KotlinVersion.CURRENT.isAtLeast(1, 5)) {
+            // Kotlin 1.4 was deprecated when this process was introduced(jackson-module-kotlin 2.15).
+            throw IllegalStateException("jackson-module-kotlin requires Kotlin 1.5 or higher.")
```

Comment:
@cowtowncoder 
If there is better error message and `exception`, I would like to hear them.

Reply:
@k163377 I think `IllegalStateException` sounds like a reasonable exception to use.

@pjfanning What exception does Scala module use for its checks (wrt compatible `jackson-databind`)? Might be useful to compare.

Reply:
Scala module uses JsonMappingException for this with a message like: 

Scala module 2.14.1 requires Jackson Databind version >= 2.14.0 and < 2.15.0 - Found jackson-databind version 2.13.2-1

Reply:
I would certainly prefer that a checked exception is used as opposed to a RuntimeException like IllegalStateException. Security 'researchers' will eventually be on the case complaining about the use of RuntimeExceptions.

Reply:
Yeah I have bit mixed feelings here. Sec research will not find this unless their systems are misconfigured -- this doesn't depend on generated input but dependencies of projects -- but yeah app developers might prefer checked exceptions.

Reply:
@pjfanning @cowtowncoder 
Thanks for the review, I pushed for the change.
2a09127